crates/holochain/src/core/ribosome/host_fn/agent_info.rs

```diff
@@ -13,8 +13,12 @@ pub fn agent_info<'a>(
     _input: AgentInfoInput,
 ) -> RibosomeResult<AgentInfoOutput> {
     let agent_pubkey = tokio_safe_block_on::tokio_safe_block_forever_on(async move {
-        let lock = call_context.host_access.workspace().read().await;
-        lock.source_chain.agent_pubkey()
+        let r = {
+            let lock = call_context.host_access.workspace().read().await;
+            lock.source_chain.agent_pubkey()
+        };
+        std::mem::drop(call_context);
+        r
```

Comment:
Does this help? What I know of rust memory management says the before and after should compile into identical code... I'd hate to have to start defensively dropping things everywhere...

Reply:
Honestly I think it makes no difference I should probably remove it. Somehow instance isn't being dropped so the arc is being kept alive but this probably isn't the location that it's happening.